pkg/interfaces/contracts/liquidity-mining/IStakelessGaugeController.sol

```diff
@@ -0,0 +1,80 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IGaugeAdder.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IStakelessGauge.sol";
+
+/**
+ * @title Stakeless Gauge Controller interface
+ * @notice Manages checkpoints for registered stakeless gauges, allowing to perform mutiple checkpoints in a
+ * single call.
+ * @dev Supports Polygon, Arbitrum, Optimism, Gnosis and ZKSync stakeless root gauges. Gauges to be checkpointed need
+ * to be registered (added) to the controller beforehand.
+ */
+interface IStakelessGaugeController {
+
+    /**
+    * @dev Emitted when stakeless gauges are added to the controller.
+    */
+    event GaugesAdded(IGaugeAdder.GaugeType gaugeType, IStakelessGauge[] gauges);
+
+    /**
+    * @dev Emitted when stakeless gauges are removed from the controller.
+    */
+    event GaugesRemoved(IGaugeAdder.GaugeType gaugeType, IStakelessGauge[] gauges);
+
+    /**
+    * @dev Registers an array of gauges from the given type.
+    * @param gaugeType - Type of the gauge.
+    * @param gauges - Gauges to register.
+    */
+    function addGauges(IGaugeAdder.GaugeType gaugeType, IStakelessGauge[] calldata gauges) external;
+
+    /**
+    * @dev De-registers an array of gauges from the given type.
+    * @param gaugeType - Type of the gauge.
+    * @param gauges - Gauges to de-register.
+    */
+    function removeGauges(IGaugeAdder.GaugeType gaugeType, IStakelessGauge[] calldata gauges) external;
+
+    /**
+    * @dev Returns true if the given gauge was registered for the given type; false otherwise.
+    * @param gaugeType - Type of the gauge.
+    * @param gauge - Gauge to check.
+    */
+    function hasGauge(IGaugeAdder.GaugeType gaugeType, IStakelessGauge gauge) external view returns (bool);
+
+    /**
+    * @dev Returns the amount of registered gauges for a given type.
+    * @param gaugeType - Type of the gauge.
+    */
+    function getTotalGauges(IGaugeAdder.GaugeType gaugeType) external view returns (uint256);
+
+    /**
+    * @dev Returns the gauge of a given type at the given index.
+    * Reverts if the index is greater than or equal to the amount of registered gauges for the given type.
+    * @param gaugeType - Type of the gauge.
+    * @param index - Index of the registered gauge.
+    */
+    function getGaugeAt(IGaugeAdder.GaugeType gaugeType, uint256 index) external view returns (address);
+
+    /**
+    * @dev Performs a checkpoint for all registered gauges above the given relative weight threshold.
+    * Reverts if the ETH sent in the call is not enough to cover bridge costs.
```

Comment:
Let's then have a getter that returns what these costs are for a given `minRelativeWeight`

Reply:
Done.
As discussed offline, this getter shares some code with `checkpointGauges` without reusing it, but this is the best approach so as not to make the code more cumbersome unnecesarily.

Reply:
We should also add it to the interface.